internal/services/nginx/nginx_deployment_data_source.go

```diff
@@ -261,12 +291,28 @@ func (m DeploymentDataSource) Read() sdk.ResourceFunc {
 					}
 
 					if scaling := props.ScalingProperties; scaling != nil {
-						output.Capacity = pointer.ToInt64(props.ScalingProperties.Capacity)
+						if capacity := scaling.Capacity; capacity != nil {
+							output.Capacity = pointer.ToInt64(props.ScalingProperties.Capacity)
+						}
+						if autoScaleProfiles := scaling.AutoScaleSettings; autoScaleProfiles != nil {
+							profiles := autoScaleProfiles.Profiles
+							for _, profile := range profiles {
+								output.AutoScaleProfile = append(output.AutoScaleProfile, AutoScaleProfile{
+									Name: profile.Name,
+									Min:  profile.Capacity.Min,
+									Max:  profile.Capacity.Max,
+								})
+							}
+						}
 					}
 
 					if userProfile := props.UserProfile; userProfile != nil && userProfile.PreferredEmail != nil {
 						output.Email = pointer.ToString(props.UserProfile.PreferredEmail)
 					}
+
+					if autoUpgradeProfile := props.AutoUpgradeProfile; autoUpgradeProfile != nil {
```

Comment:
Pretty sure we just need to check for `props.UpgradeChannel` to not be nil and set `output.UpgradeChannel = props.UpgradeChannel`